manager/allocator/network.go

```diff
@@ -1052,6 +1052,9 @@ func (a *Allocator) allocateNode(ctx context.Context, node *api.Node, existingAd
 		// for every attachment, go through every network. if the attachment
 		// belongs to one of the networks, then go to the next attachment. if
 		// no network matches, then the the attachment should be removed.
+		if attach.Network == nil {
```

Comment:
How can this happen? When attaching a network that's not properly working (eg conflicts with another subnet)? Shouldn't we remove it from the node's attachments then?

In any case, seems worth at least a comment :)

Reply:
+1 for comment.

Reply:
yes, that's great point, but may be we should just return error and fail operation?